Actions updates

tdonaworth opened the pull request:
* Created a few example Github Action workflows to consume
* Created a TFSec Action to consume
* Closes #5 Recommendations on Security Analysis of Infrastructure as Code Workflows

tdonaworth committed:
initial commit

tdonaworth committed:
formatting fix

tdonaworth committed:
updated example uses

tohch4 commented on actions/README.md:
```diff
@@ -0,0 +1,143 @@
+
+## Terraform
+### TFSec
+`actions/tfsec-action` will run the [Flexion/tfsec-action](https://github.com/flexion/flexion-sig-security/actions/tfsec-action) on all `pull_request` for the repositories. This simply runs the `tfsec` docker container from within an action. 
```
I think if we do the path method like this with just `./path` and not the full URL it will work in GitHub and local repo on your computer in an IDE like VS Code, regardless of what branch you are in.

[Example from my other projects](https://github.com/GSA/fedramp-automation/blob/74b3db7b580afe4610f3a18d9dbfec2d9abba4b7/src/validations/README.md#developers-need-help-or-have-a-technical-question) and our Markdown docs. We determined `../up_a_level_docs.md` work very nicely too.

```suggestion
`actions/tfsec-action` will run the [Flexion/tfsec-action](./actions/tfsec-action) on all `pull_request` for the repositories. This simply runs the `tfsec` docker container from within an action. 
```

tohch4 commented on actions/README.md:
```diff
@@ -0,0 +1,143 @@
+
+## SonarQube/Cloud
+### SonarQube
+[SonarSource/sonarqube-scan-action](https://github.com/SonarSource/sonarqube-scan-action) is an officially supported action for running the SonarQube CLI Scanner within Github Actions.
```
```suggestion
[SonarSource/sonarqube-scan-action](https://github.com/SonarSource/sonarqube-scan-action) is an officially supported action for running the SonarQube CLI Scanner within GitHub Actions.
```

tohch4 commented on actions/README.md:
```diff
@@ -0,0 +1,143 @@
+</details>
+
+### SonarCloud
+[SonarSource/sonarcloud-github-action](https://github.com/SonarSource/sonarcloud-github-action) is an officially supported action for running the SonarCloud Scanner within Github Actions.
```
```suggestion
[SonarSource/sonarcloud-github-action](https://github.com/SonarSource/sonarcloud-github-action) is an officially supported action for running the SonarCloud Scanner within GitHub Actions.
```

tohch4 commented on actions/tfsec-action/README.md:
```diff
@@ -0,0 +1,36 @@
+# TFSEC Action
+
+This github action scans a repository usuing AquaSecurity's [TFSEC](https://github.com/aquasecurity/tfsec) tool.
```
```suggestion
This GitHub action scans a repository using AquaSecurity's [tfsec](https://github.com/aquasecurity/tfsec) tool.
```

tohch4 commented on actions/tfsec-action/action.yml:
```diff
@@ -0,0 +1,23 @@
+name: 'Run TFSEC'
+description: >
+  Runs the TFSEC tool, to check Terraform for Security issues. 
+branding:
+  icon: 'git-pull-request'
+  color: 'purple'
```
```suggestion
name: 'Run tfsec'
description: >
  Runs the tfsec tool, to check Terraform for Security issues. 
branding:
  icon: 'git-pull-request'
  color: 'purple'
```

tohch4 reviewed (changes requested):
I have some minor feedback on capitalization and typos.

Something more systematic: should we not write a small blurb, or lead by example, and pin to specific versions of a GitHub action (either SHA1 hash or tag, but I think maybe the former is worth considering) and review updates to the action bundles themselves?

This guidance was encouraged by peers in a Flexion project for federal agency customers, and I thought the internal guidance document is sound; GitHub itself, however, does not expressly provide that [in their hardening guidance](https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions).

**EDIT:** There are [a few articles like this supporting my recommendation here](https://julienrenaux.fr/2019/12/20/github-actions-security-risk/), and I cannot locate the one that matters: someone was able to compromise the GitHub Actions infrastructure with this technique or chaining it [because of how GitHub's actions organization works](https://github.com/actions/). If I can find the article from my personal workstation later today, I shall post it.

Thoughts?

tdonaworth committed:
Update actions/README.md

Co-authored-by: Alexander Stein <39844334+tohch4@users.noreply.github.com>

tdonaworth committed:
Update actions/tfsec-action/README.md

Co-authored-by: Alexander Stein <39844334+tohch4@users.noreply.github.com>

tdonaworth committed:
Update actions/tfsec-action/action.yml

Co-authored-by: Alexander Stein <39844334+tohch4@users.noreply.github.com>

tdonaworth committed:
Update actions/README.md

Co-authored-by: Alexander Stein <39844334+tohch4@users.noreply.github.com>

tdonaworth committed:
Update actions/README.md

Co-authored-by: Alexander Stein <39844334+tohch4@users.noreply.github.com>

tdonaworth committed:
updated semantic versions of actions

tdonaworth committed:
Merge branches 'actions-updates' and 'actions-updates' of github.com:flexion/flexion-sig-security into actions-updates

tdonaworth commented:
* Updated the referenced versions of the actions within the examples. 
* Updated the referenced version of tfsec docker container.
Have no current version to reference for the `actions/tfsec-action` - we should either do a release once committed or pull it out into it's own repo to track separately.
